client/client-lib/src/mint/mod.rs

```diff
@@ -155,6 +153,110 @@ impl MintClient {
     }
 
+    /// Adds the final amounts of `change` to the tx before submitting it
+    /// Allows for multiple `change` outputs
+    pub async fn finalize_change(&self, tx: &mut Transaction, change: Vec<Amount>) {
+        let mut change_outputs: Vec<(usize, NoteIssuanceRequests)> = vec![];
+
+        for amount in change {
+            let (issuances, nonces) = self.create_output_coins(amount).await;
+            let out_idx = tx.outputs.len();
+            tx.outputs.push(Output::Mint(MintOutput(nonces)));
+            change_outputs.push((out_idx, issuances));
+        }
+        let txid = tx.tx_hash();
+
+        let mut dbtx = self.context.db.begin_transaction(ModuleRegistry::default());
+
+        // remove the spent ecash from the DB
+        let mut input_ecash: Vec<(Amount, SpendableNote)> = vec![];
+        for input in &tx.inputs {
+            if let Input::Mint(MintInput(notes)) = input {
+                for (amount, note) in notes.clone() {
+                    let key = CoinKey {
+                        amount,
+                        nonce: note.0,
+                    };
+                    let spendable = dbtx
+                        .get_value(&key)
+                        .await
+                        .expect("DB Error")
+                        .expect("Missing note");
+                    input_ecash.push((amount, spendable));
+                    dbtx.remove_entry(&key).await.expect("DB Error");
+                }
+            }
+        }
+
+        // move ecash to pending state, awaiting a transaction
+        if !input_ecash.is_empty() {
+            let pending = TieredMulti::from_iter(input_ecash.into_iter());
+            dbtx.insert_entry(&PendingCoinsKey(txid), &pending)
+                .await
+                .expect("DB Error");
+        }
+
+        // write ecash outputs to db to await for tx success to be fetched later
+        for (out_idx, coins) in change_outputs.iter() {
+            dbtx.insert_new_entry(
+                &OutputFinalizationKey(OutPoint {
+                    txid,
+                    out_idx: *out_idx as u64,
+                }),
+                &coins.clone(),
+            )
+            .await
+            .expect("DB Error");
+        }
+
+        dbtx.commit_tx().await.expect("DB Error");
+    }
+
+    async fn create_output_coins(
+        &self,
+        amount: Amount,
+    ) -> (NoteIssuanceRequests, TieredMulti<BlindNonce>) {
+        let mut amount_requests: Vec<((Amount, NoteIssuanceRequest), (Amount, BlindNonce))> =
+            Vec::new();
+        for (amt, ()) in TieredMulti::represent_amount(amount, &self.config.tbs_pks).into_iter() {
+            let (request, blind_nonce) = self.new_ecash_note(&self.context.secp, amt).await;
+            amount_requests.push(((amt, request), (amt, blind_nonce)));
+        }
+        let (coin_finalization_data, sig_req): (NoteIssuanceRequests, MintOutput) =
+            amount_requests.into_iter().unzip();
```

Comment:
nit: we could push directly into different data structures instead of one vec and then calling unzip.